protocols/gossipsub/src/behaviour.rs

```diff
@@ -2152,18 +2154,23 @@ where
             let mesh_outbound_min = self.config.mesh_outbound_min_for_topic(topic_hash);
 
             // drop all peers with negative score, without PX
-            // if there is at some point a stable retain method for BTreeSet the following can be
-            // written more efficiently with retain.
-            let mut to_remove_peers = Vec::new();
-            for peer_id in peers.iter() {
+            // report peers below the score report threshold.
+            let mut removed_peers = 0;
```

Comment:
Is it a good idea to change `to_remove_peer` in this PR? It doesn't seem related